pkg/cmd/operator/webhooks.go

```diff
@@ -91,9 +93,37 @@ type Validator interface {
 	GetWarningsOnUpdate(obj, oldObj runtime.Object) []string
 }
 
+// Container runtime may set env var (SCYLLA_OPERATOR_PORT) to either number or "tcp://<service-host>:<service-port>"
+var portFlagRe = regexp.MustCompile(`^(?:tcp://[^:]+:|)(\d+)$`)
```

Comment:
nit: `^(?:tcp://[^:]+:)?(\d+)$` seems clearer to me (optional group marked with `?`), but that's up to you